lizmap/modules/lizmap/classes/lizmapOGCRequest.class.php

```diff
@@ -20,17 +20,75 @@ class lizmapOGCRequest
 
     protected $params;
 
+    protected $requestXml;
+
     protected $services;
 
     protected $tplExceptions;
 
+    static public function build($project, $params, $requestXml = Null)
+    {
+        $service = Null;
+        $request = Null;
+
+        // Check request XML
+        if ($requestXml && substr(trim($requestXml), 0, 1) == '<') {
+            $requestXml = trim($requestXml);
+        } else {
+            $requestXml = Null;
+        }
+
+        // Parse request XML
+        if ($requestXml) {
+            $xml = simplexml_load_string($requestXml);
+            if ($xml) {
+                $request = $xml->getName();
+                if (property_exists($xml->attributes(), 'service')) {
+                    $service = $xml['service'];
+                }
+            } else {
+                $requestXml = Null;
+            }
+        }
+
+        // Check parameters
+        if (!$requestXml && isset($params['service'])) {
+            $service = strtoupper($params['service']);
```

Comment:
Could you explain why you use `strtoupper()` now ?

Reply:
OGC service has to be upper case for QGIS Server, I will add a comment